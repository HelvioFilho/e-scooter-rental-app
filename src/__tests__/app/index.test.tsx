import React from "react";
import { render } from "@testing-library/react-native";
import Home from "@/app/index";

jest.mock("expo-router", () => ({
  Stack: {
    Screen: jest.fn(() => null),
  },
}));

jest.mock("@/components/Map", () => {
  const { View } = require("react-native");
  return jest.fn(() => <View testID="map" />);
});

jest.mock("@/components/SelectedScooterSheet", () => {
  const { View } = require("react-native");
  return jest.fn(() => <View testID="selected-scooter-sheet" />);
});

describe("Home Screen", () => {
  it("should render the StatusBar, Stack.Screen, Map, and SelectedScooterSheet components", () => {
    const { getByTestId, queryByText } = render(<Home />);

    expect(getByTestId("map")).toBeTruthy();
    expect(getByTestId("selected-scooter-sheet")).toBeTruthy();
    expect(queryByText("Home")).toBeNull();
  });

  it("should match the snapshot", () => {
    const tree = render(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
