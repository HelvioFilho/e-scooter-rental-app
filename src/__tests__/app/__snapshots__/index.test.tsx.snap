// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen should match the snapshot 1`] = `
[
  <View
    testID="map"
  />,
  <View
    testID="selected-scooter-sheet"
  />,
]
`;
